packages/frontend-web/src/app/scenes/Settings/components/OAuthConfig.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2019 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useCallback, useState } from 'react';
+
+import {
+  Button,
+  makeStyles,
+  TextField,
+} from '@material-ui/core';
+
+import { ContainerHeader, OverflowContainer } from '../../../components/OverflowContainer';
+import { Scrim } from '../../../components/Scrim';
+import { API_URL } from '../../../config';
+import { IApiConfiguration } from '../../../platform/dataService';
+import { SCRIM_STYLE } from '../../../styles';
+import { css } from '../../../utilx';
+
+export interface IOAuthConfigProps extends IApiConfiguration {
+  onClickDone(config: IApiConfiguration): any;
+  showLogoutWarning?: boolean;
+}
+
+const useStyles = makeStyles((_theme) => ({
+  textField: {
+    width: '40vw',
+  },
+}));
+
+export function OAuthConfig(props: IOAuthConfigProps) {
+  const [id, setId] = useState(props.id);
+  const [secret, setSecret] = useState(props.secret);
+  const classes = useStyles({});
+
+  const idHandler = useCallback((e) => setId(e.target.value), [setId]);
+  const secretHandler = useCallback((e) => setSecret(e.target.value), [setSecret]);
+  function submitHandler() {
+    props.onClickDone({id, secret});
+  }
+
+  return (
+    <div>
+      <p>Configure your connection to the Google Authentication service.</p>
+      <p>
+        To allocate and view tokens, visit <a
+          href="https://console.developers.google.com/apis/credentials"
+          target="_blank"
+          style={{color: 'inherit', fontWeight: 'bold', textDecoration: 'underline'}}
+        >the Google API console</a>.
+        If you haven't yet got an appropriate <b>OAuth Client ID</b> yet, or want to allocate a new one,
```

Comment:
haven't yet got doesn't scan well. and there's two yets. suggest:
If you don't yet have an OAuth Client ID

Reply:
That is way better.  Thanks for that.